backend/controllers/AuthController.ts

```diff
@@ -50,6 +52,39 @@ const AuthController = {
       return res.status(500).json({ error: err.message });
     }
   },
+  async forgotPassword(req: Request, res: Response) {
+    try {
+      const { email } = req.body;
+
+      const user = await UserModel.findOne({ email });
+
+      if (!user) {
+        return res.status(400).json({
+          message: 'There is no user with this email address...',
+          success: false,
+        });
+      }
+
+      const newPassword = uuidv4();
+      user.passwordHash = newPassword;
```

Comment:
We don't have to hash the password since we already have Mongoose hook doing it.
``backend/models/UserModel.ts``
```js
schema.pre("save", async function (next) {
  const user: IUser = this;
  const hash = await bcrypt.hash(user.passwordHash, 10);

  user.passwordHash = hash;
  next();
});
```